### PR TITLE
Live data

### PR DESCRIPTION
**Pull request by ChoiJ0527**
The React-stockcharts component now displays basic candlesticks using live data that was queried from the backend server.

- from: begin minus 5097600 (2 months in UNIX time format)
- begin: today
- ticker: currently selected ticker
- resolution: 30 minute interval

The React-stockcharts component has been modified to display hour and minute for the mouse coordinate X timestamp.
The current price now displays the last close value from the queried data.
This page now shows the message "Data error, please try a different ticker." if the server query returns false for hasData.

### CODE REVIEW

File: client/src/components/Chart copy.js [hunk 1]
```diff
@@ -0,0 +1,299 @@
+import { format } from "d3-format";
+import { timeFormat } from "d3-time-format";
+
+import React from "react";
+import PropTypes from "prop-types";
```
Comment (willyspinner) [inline client/src/components/Chart copy.js:5]: Is this file even used? I don't see it anywhere.  If not then you should delete it.

File: client/src/components/Chart pan copy.js [hunk 2]
```diff
@@ -0,0 +1,301 @@
+import { format } from "d3-format";
```
Comment (willyspinner) [inline client/src/components/Chart pan copy.js:1]: This file too. This is unused

File: client/src/pages/chart copy.jsx [hunk 3]
```diff
@@ -0,0 +1,235 @@
+import React, { Component } from 'react';
```
Comment (willyspinner) [inline client/src/pages/chart copy.jsx:1]: this file also seems unused.  Pls delete unnecessary files

File: client/src/pages/chart.jsx [hunk 4]
```diff
@@ -17,24 +17,30 @@ import { getPortfolioItems } from '../services/portfolioService';
 import Portfolio from '../components/Portfolio';
 import StockSelector from '../components/StockSelector';
 import '../styles/app.css';
-class ChartComponent extends Component {
+import { timeParse } from "d3-time-format";
 
+const parseDate = timeParse("%s");
+
+class ChartComponent extends Chart {
```
Comment (willyspinner) [inline client/src/pages/chart.jsx:24]: Why are we extending the `Chart` component here?  For what reason?

File: client/src/utils.js [hunk 5]
```diff
@@ -15,14 +18,12 @@ function parseData(parse) {
...
 
-export function getData() {
-	const promiseMSFT = fetch("https://cdn.rawgit.com/rrag/react-stockcharts/master/docs/data/MSFT.tsv")
-		.then(response => response.text())
-		.then(data => tsvParse(data, parseData(parseDate)))
-	return promiseMSFT;
+const apiEndpoint = apiUrl + "/chart";
 
+export function getData(begin, end, ticker, resolution) {
+	return http.get(apiEndpoint + "?from=" + begin + "&to=" + end + "&ticker=" + ticker + "&resolution=" + resolution);
```
Comment (willyspinner) [inline client/src/utils.js:26]: You need to follow our conventions and define this in a `....service.js` file. 

This should be in a new file called `services/chartingService.js`, not in utils.

File: client/src/utils.js [hunk 6]
```diff
@@ -1,9 +1,12 @@
-import { tsvParse } from  "d3-dsv";
+import http from "./services/httpService";
```
Comment (willyspinner) [inline client/src/utils.js:1]: as per comment below, `http` shouldn't be imported here. It should be in the service file.

**Review by willyspinner — CHANGES_REQUESTED** [review]
Nice work 👍 

You need to cleanup your branch and then we can merge this

File: client/src/pages/chart.jsx [hunk 7]
```diff
@@ -91,7 +101,33 @@ class ChartComponent extends Component {
...
   }
 
-  
+  runGetData(ticker) {
+    getData(this.state.end - 5097600, this.state.end, ticker, "30").then(({ data }) => {
+      this.setState({ data })
+      this.setState({ hasQueryData : data.hasData})
+      if (data.hasData)
+      {
+        this.setState({ mostRecentPrice : data.closes[data.closes.length - 1] });
```
Comment (willyspinner) [inline client/src/pages/chart.jsx:110]: Why not put all this in a single `setState` call?

```
this.setState(prevState=> ({
  data,
  hasQueryData: data.hasData,
  mostRecentPrice: data.hasData? data.closes[data.closes.length - 1] : prevState.mostRecentPrice
}));

File: client/src/components/Chart copy.js (hunk 1 above)
Comment (ChoiJ0527) [reply client/src/components/Chart copy.js]: deleted

File: client/src/components/Chart pan copy.js (hunk 2 above)
Comment (ChoiJ0527) [reply client/src/components/Chart pan copy.js]: deleted

File: client/src/pages/chart.jsx (hunk 4 above)
Comment (ChoiJ0527) [reply client/src/pages/chart.jsx]: Remnant of old attempt to get this working, changed it back to extend Component.

File: client/src/pages/chart copy.jsx (hunk 3 above)
Comment (ChoiJ0527) [reply client/src/pages/chart copy.jsx]: Deleted

File: client/src/utils.js (hunk 5 above)
Comment (ChoiJ0527) [reply client/src/utils.js]: Done

File: client/src/utils.js (hunk 6 above)
Comment (ChoiJ0527) [reply client/src/utils.js]: Done

File: client/src/pages/chart.jsx (hunk 7 above)
Comment (ChoiJ0527) [reply client/src/pages/chart.jsx]: Done

**Review by willyspinner — APPROVED** [review]

**Pull request merged**